primitives/src/v2/mod.rs

```diff
@@ -1569,6 +1569,78 @@ impl CompactStatement {
 	}
 }
 
+/// Validators struct indexed by ValidatorIndex.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq, MallocSizeOf))]
+pub struct Validators(Vec<ValidatorId>);
+
+impl Index<ValidatorIndex> for Validators {
+	type Output = ValidatorId;
+
+	fn index(&self, index: ValidatorIndex) -> &Self::Output {
+		&self.0.get(index.0 as usize).unwrap()
```

Comment:
This looks dangerous. Should `Output` be option ?

Reply:
I implemented it just in case. 
Seen a few places where the original were indexed using e.g. `validators[0]` syntax

Reply:
>  Should Output be option ?

Tried this approach, couldn't quite hack it

Reply:
Agree this looks as dangerous as using `validators[index.0 as usize]` which panics for a vector on out-of-bounds access.

I'd prefer to remove this implementation completely and replace the usage of indexing with `get(index).unwrap()` in tests and `get(index).expect("<write an actual proof here; qed>")` elsewhere.